src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/ParseJson.txt

```diff
@@ -224,3 +224,7 @@ Blank()
 
 >> Text(Index(ParseJSON("[ { ""a"" : 10 }, { ""b"" : ""str"" } ]"),2).a)
 Blank()
+
+// ForAll
+>> ForAll(ParseJSON("[5, 6, 7]"), Value(Value))
+[5,6,7]
```

Comment:
Some interesting cases to try (the examples below use ThisRecord, but change it to using Value if we decide to go that route):
- Expression returns an untyped object: `ForAll(ParseJSON("[1,2,3]"), ThisRecord)`
- Dealing with heterogeneous arrays:

    ForAll(
      ParseJSON("[1,{""a"":false},""hello""]),
      IfError(
        Text(ThisRecord),
        IfError(
          Text(Value(ThisRecord)),
          Text(Boolean(a))
        )
      )
    )

- Error cases (error should be returned): `ForAll(ParseJSON("[1,true,3,false]"), Value(ThisRecord))`

Reply:
Added those tests